src/shared/utils/rateLimitDetector.ts

```
@@ -10,3 +10,153 @@ const RATE_LIMIT_SUBSTRING = "You've hit your limit";
+}
+
+// ---------------------------------------------------------------------------
+// Absolute clock times: "resets at 3pm (PST)", "resets at 15:30 UTC"
+// ---------------------------------------------------------------------------
+
+/**
+ * Captures the clock time + optional timezone abbreviation from phrases like
+ * "reset at 3pm (PST)" or "resets at 15:30 UTC".
+ */
+const ABSOLUTE_RESET_RE =
+  /reset(?:s|ting)?\s+(?:at\s+)?(\d{1,2})(?::(\d{2}))?\s*(am|pm)?\s*(?:\(([A-Z]{2,5})\)|([A-Z]{2,5}))?/i;
```

ESLint

[error] Simplify this regular expression to reduce its complexity from 27 to the 20 allowed.

(sonarjs/regex-complexity)